### PR TITLE
mj-snow-filtersandAttributeCleanup

### PR DESCRIPTION
**Pull request by marc-sgnl**
Tested both templates in SGNL, ingested data for all attributes successfully, no errors.

https://app.shortcut.com/sgnl/story/25184/cleanup-of-snow-templates

### CODE REVIEW

File: servicenow-itsm.sgnl.yaml [hunk 1]
```diff
@@ -9,8 +9,20 @@ defaultApiCallFrequency: SECONDLY
 defaultApiCallMinInterval: 1
 type: "ServiceNow-1.0.1"
 # Example Config:
-# {"requestTimeoutSeconds": 10, "apiVersion": "v2"}
-adapterConfig: "eyJyZXF1ZXN0VGltZW91dFNlY29uZHMiOiAxMCwgImFwaVZlcnNpb24iOiAidjIifQ=="
+#{
+#  "requestTimeoutSeconds": 10,
+#  "apiVersion": "v2",
+#  "filters": {
+#    "incident": "active=true",
```
Comment (r-rakshith) [inline servicenow-itsm.sgnl.yaml:16]: nit: We need not update the default adapter config but it'd be nice for the users to have some filtering syntax example with a helpful filter

```suggestion
#    "incident": "active=true^sys_updated_onRELATIVEGE@minute@ago@60",
```

**Review by r-rakshith — APPROVED** [review]

File: servicenow-itsm.sgnl.yaml (hunk 1 above)
Comment (marc-sgnl) [reply servicenow-itsm.sgnl.yaml:16]: Oh I like that, I will in fact add it to the config too, I think - from https://docs.servicenow.com/bundle/washingtondc-platform-user-interface/page/use/common-ui-elements/reference/r_OpAvailableFiltersQueries.html

I believe, "Any Incident updated in the last 30 days" is:

"incident": "active=true^sys_updated_onRELATIVEGT@day@ago@30"

Is that your understanding too?

Comment (r-rakshith) [reply servicenow-itsm.sgnl.yaml]: That looks correct to me. I'd recommend we also test this filter against a live instance before we publish this.

File: servicenow-csm.sgnl.yaml [hunk 2]
```diff
@@ -9,8 +9,17 @@ defaultApiCallFrequency: SECONDLY
 defaultApiCallMinInterval: 1
 type: "ServiceNow-1.0.1"
 # Example Config:
-# {"requestTimeoutSeconds": 10, "apiVersion": "v2"}
-adapterConfig: "eyJyZXF1ZXN0VGltZW91dFNlY29uZHMiOiAxMCwgImFwaVZlcnNpb24iOiAidjIifQ=="
+#{
```
Comment (r-rakshith) [inline servicenow-csm.sgnl.yaml:12]: nit
```suggestion
# {
```

File: servicenow-csm.sgnl.yaml [hunk 3]
```diff
@@ -9,8 +9,17 @@ defaultApiCallFrequency: SECONDLY
...
+#  "requestTimeoutSeconds": 10,
+#  "apiVersion": "v2",
+#  "filters": {
+#    "sn_customerservice_case": "active=true^sys_updated_onRELATIVEGT@dayofweek@ago@30",
+#    "sys_user": "active=true",
+#    "customer_account": "active=true",
+#    "sys_user_group": "active=true"
+#  }
+#}
+adapterConfig: "ewogICJyZXF1ZXN0VGltZW91dFNlY29uZHMiOiAxMCwKICAiYXBpVmVyc2lvbiI6ICJ2MiIsCiAgImZpbHRlcnMiOiB7CiAgICAic25fY3VzdG9tZXJzZXJ2aWNlX2Nhc2UiOiAiYWN0aXZlPXRydWVec3lzX3VwZGF0ZWRfb25SRUxBVElWRUdUQGRheUBhZ29AMzAiLAogICAgInN5c191c2VyIjogImFjdGl2ZT10cnVlIiwKICAgICJjdXN0b21lcl9hY2NvdW50IjogImFjdGl2ZT10cnVlIiwKICAgICJzeXNfdXNlcl9ncm91cCI6ICJhY3RpdmU9dHJ1ZSIKICB9Cn0="
```
Comment (r-rakshith) [inline servicenow-csm.sgnl.yaml:22]: This config uses `@day` whereas the comment uses `@dayofweek`. Please update this.

**Review by r-rakshith — CHANGES_REQUESTED** [review]
Requesting fix for one small type.

In general it is confusing on whether `@day` is correct or `@dayofweek`. I see conflicting examples online.

`@dayofweek`: [a post](https://www.servicenow.com/community/knowledge-management-forum/please-help-with-modifying-the-search-source-code-of-knowledge/m-p/2625009#:~:text=%22more_ago%22%3A%C2%A0%22%5Esys_updated_onRELATIVEGT,dayofweek%40ago%401%22%2C)
`@day`: `sla_dueSAMEASactivity_due@day`, `sla_dueMORETHANactivity_due@day@before@1`, `sla_dueLESSTHANactivity_due@day@before@3` in [SNoW docs](https://docs.servicenow.com/bundle/washingtondc-platform-user-interface/page/use/common-ui-elements/reference/r_OpAvailableFiltersQueries.html)


To ensure correctness, can we please try these filters out in the UI and export them to a query as shown in [this post](https://www.servicenow.com/community/developer-forum/how-to-query-incidents-in-servicenow-which-updated-or-created-in/m-p/2101045/highlight/true#M757971)

File: servicenow-csm.sgnl.yaml (hunk 3 above)
Comment (marc-sgnl) [reply servicenow-csm.sgnl.yaml:22]: @dayofweek is correct as tested in the console, will update the config.

File: servicenow-itsm.sgnl.yaml (hunk 1 above)
Comment (marc-sgnl) [reply servicenow-itsm.sgnl.yaml]: Ah yeah, sorry didnt update this comment -- tried day first, didn't work, switched over to dayofweek

**Review by r-rakshith — APPROVED** [review]
Thank you!

**Pull request merged**